SonarRunner.Shim/PropertiesWriter.cs

```diff
@@ -124,16 +124,21 @@ public void WriteSettingsForProject(ProjectInfo project, IEnumerable<string> fil
 
             if (fxCopReportFilePath != null)
             {
-                string property;
-                if (projectLanguage == ProjectLanguage.CS)
+                string property = null;
+                if (ProjectLanguages.IsCSharpProject(project.ProjectLanguage))
                 {
                     property = "sonar.cs.fxcop.reportPath";
                 }
-                else
+                else if (ProjectLanguages.IsVbProject(project.ProjectLanguage))
                 {
                     property = "sonar.vbnet.fxcop.reportPath";
                 }
-                AppendKeyValue(sb, guid, property, fxCopReportFilePath);
+
+                Debug.Assert(property != null, "FxCopReportFilePath is set but the language is unrecognised. Language: " + project.ProjectLanguage);
+                if (property != null)
+                {
+                    AppendKeyValue(sb, guid, property, fxCopReportFilePath);
+                }
```

Comment:
I've added a test to check that specify an FxCop report for an unknown language won't emit anything into the project file.